src/Tasks/Microsoft.NET.Build.Tasks/GenerateGlobalUsings.cs

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using Microsoft.Build.Framework;
+
+namespace Microsoft.NET.Build.Tasks
+{
+    public sealed class GenerateGlobalUsings : TaskBase
+    {
+        [Required]
+        public ITaskItem[] Usings { get; set; }
+
+        [Required]
+        public ITaskItem OutputFile { get; set; }
+
+        protected override void ExecuteCore()
+        {
+            if (Usings.Length == 0)
+            {
+                return;
+            }
+
+            var usings = Usings.Select(UsingInfo.Read)
+                .OrderBy(static k => k, UsingInfoComparer.Instance)
+                .Distinct(UsingInfoComparer.Instance);
+
+            StringBuilder builder = new();
+            builder.AppendLine("// <auto-generated/>");
+            foreach (var @using in usings)
+            {
+                builder.Append("global using ");
+
+                if (@using.Static)
+                {
+                    builder.Append("static ");
+                }
+
+                if (!string.IsNullOrEmpty(@using.Alias))
+                {
+                    builder.Append(@using.Alias)
+                        .Append(" = ");
+                }
+
+                builder.Append("global::")
+                    .Append(@using.Namespace)
+                    .Append(';')
+                    .AppendLine();
+            }
+
+            var contentToWrite = builder.ToString();
+            var outputFilePath = Path.GetFullPath(OutputFile.ItemSpec);
+            if (File.Exists(outputFilePath))
+            {
+                var existing = File.ReadAllText(outputFilePath);
+
+                if (string.Equals(existing, contentToWrite, StringComparison.Ordinal))
+                {
+                     Log.LogMessage(MessageImportance.Low, Strings.SkippingUnchangedFile, OutputFile.ItemSpec);
+                    return;
+                }
```

Comment:
I tried that before on my failed PR and it would result in test failures, I also tried consuming that task directly in a task like this and it would pass when ran on it's own on the specific test assembly for this dll, but then fail on other tests or at least the same tests in CI vs when ran locally.